### PR TITLE
Ghidra: Unit Test

### PR DESCRIPTION
**Pull request by colton-gabertan**
### Checklist

- [X] No CHANGELOG update needed

- [ ] No new tests needed

- [ ] No documentation update needed

# Implement Ghidra Unit Testing
---

### CODE REVIEW

**Comment by colton-gabertan** [comment]
Note 8/14:

This branch is being used to test the new Ghidrathon version, see https://github.com/mandiant/Ghidrathon/tree/fix/65

File: tests/test_ghidra_features.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        wanted_str = "pma" + str(psx_str).split("/")[-1][len("Practical Malware Analysis Lab ") : -5]
+    else:
+        # <PosixPath>/mimikatz.exe_ -> mimikatz
+        wanted_str = str(psx_str).split("/")[-1][:-5]
+
+    if "_" in wanted_str:
+        # al-khaser_x86 -> al-khaser x86
+        wanted_str = wanted_str.replace("_", " ")
+
+    return wanted_str
```
Comment (mike-hunhoff) [inline tests/test_ghidra_features.py:46]: This code looks difficult to maintain. Can we compare the provided path to `currentProgram().getExecutablePath()`? We likely need to wrap the string using `PosixPath` before comparing the two.

File: tests/test_ghidra_features.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        check_input_file(sample)
+    except RuntimeError:
+        pytest.skip(reason="Test must be ran against sample loaded in Ghidra")
+
+    fixtures.do_test_feature_count(get_ghidra_extractor, sample, scope, feature, expected)
+
+
+if __name__ == "__main__":
+    # No support for faulthandler module in Ghidrathon, see:
+    # https://github.com/mandiant/Ghidrathon/issues/70
```
Comment (mike-hunhoff) [inline tests/test_ghidra_features.py:97]: Thank you for documenting!

**Review by mike-hunhoff — CHANGES_REQUESTED** [review]
Thank you @colton-gabertan ! I've left comments for your review and opened https://github.com/mandiant/capa/issues/1732 to track the TODO mentioned in `tests.yml`.

File: tests/test_ghidra_features.py (hunk 1 above)
Comment (mike-hunhoff) [reply tests/test_ghidra_features.py:46]: Ok so it looks like we need the opposite of `fixtures.get_sample_md5_by_name` e.g. `fixtures.get_sample_short_name_by_md5` or the like. Let's leave this code as-is for now and I'll open a separate issue to update the fixtures.

File: tests/test_ghidra_features.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        raise RuntimeError(f"please run the tests against sample with MD5: `{found}`")
+
+
+def get_ghidra_extractor(path):
+    import capa.features.extractors.ghidra.extractor
+
+    extractor = capa.features.extractors.ghidra.extractor.GhidraFeatureExtractor()
+    setattr(extractor, "path", path.as_posix())
+
+    return extractor
```
[the remark this answers is not in the document]
Comment (mike-hunhoff) [reply tests/test_ghidra_features.py:70]: @colton-gabertan let's move this function to `capa/tests/fixtures.py` alongside the other `get_xx_extractor` functions unless I am missing a reason to keep as-is.

File: .github/workflows/tests.yml [hunk 4]
```diff
@@ -181,20 +181,21 @@ jobs:
         wget "https://github.com/NationalSecurityAgency/ghidra/releases/download/Ghidra_${{ matrix.ghidra-version }}_build/ghidra_${{ matrix.ghidra-version }}_${{ matrix.public-version }}.zip" -O ./.github/ghidra/ghidra_${{ matrix.ghidra-version }}_PUBLIC.zip
         unzip .github/ghidra/ghidra_${{ matrix.ghidra-version }}_PUBLIC.zip -d .github/ghidra/
     - name: Install Ghidrathon
-      run : |
+      run : | # Colton TODO: fix wget url & unzip paths pending merge of breaking Ghidrathon changes
```
Comment (mike-hunhoff) [inline .github/workflows/tests.yml:184]: I have released https://github.com/mandiant/Ghidrathon/releases/tag/v3.0.0 with these breaking changes. Let's update the code here to reflect that.

File: tests/test_ghidra_features.py (hunk 1 above)
Comment (colton-gabertan) [reply tests/test_ghidra_features.py:46]: Example of what's passed in the `sample` parameter for fixtures:
`sample = PosixPath('/home/wumbo/capa/tests/data/mimikatz.exe_')`

File: .github/workflows/tests.yml [hunk 5]
```diff
@@ -194,7 +194,5 @@ jobs:
     - name: Install capa
       run: pip install -e .[dev] 
     - name: Run tests
-      run: | # runs main.py for now...
-        .github/ghidra/ghidra_${{ matrix.ghidra-version }}_PUBLIC/support/analyzeHeadless .github/ghidra/project ghidra_test -Import ./tests/data/'Practical Malware Analysis Lab 01-01.dll_'
-        .github/ghidra/ghidra_${{ matrix.ghidra-version }}_PUBLIC/support/analyzeHeadless .github/ghidra/project ghidra_test -process 'Practical Malware Analysis Lab 01-01.dll_' -ScriptPath ./capa -PostScript main.py
+      run: .github/ghidra/ghidra_${{ matrix.ghidra-version }}_PUBLIC/support/analyzeHeadless .github/ghidra/project ghidra_test -Import ./tests/data/'mimikatz.exe_' -ScriptPath ./tests/ -PostScript test_ghidra_features.py
```
Comment (mike-hunhoff) [inline .github/workflows/tests.yml:197]: ❤️

**Pull request merged**